share/functions/__fish_print_pipestatus.fish

```diff
@@ -1,19 +1,35 @@
 function __fish_print_pipestatus --description "Print pipestatus for prompt"
-    # maybe these could be in global variables similar to __fish_color_status to allow
-    # users to set the variables to modify the braces/separator/color used
-    set -l left_brace $argv[1]
-    set -l right_brace $argv[2]
-    set -l separator $argv[3]
-    set -l brace_sep_color $argv[4]
-    set -l status_color $argv[5]
+
+    # allow users to customize these variables
+    if not set -q __fish_prompt_pipestatus_left_brace
```

Comment:
I'd leave "left_brace" and "right_brace" out of this - the calling function can already do that via

```fish
'['(__fish_print_pipestatus)']'
```

Reply:
Also I'm not a massive fan of global variables for this. The main place to start customizing the prompt is the fish_prompt function. Finding the call there and changing the arguments is easier than finding the variable names and setting those, possibly somewhere outside of the prompt.

Reply:
Thank you, I reverted the global variables.

Regarding leaving `left_brace` and `right_brace` how would I avoid outputting them if there was no pipe or all processes of the pipe succeeded? The `$pipestatus` values are only shown in the prompt if there was an error in any process that made up the pipe. This is similar to how `$status` is only displayed if a command failed.

Reply:
> how would I avoid outputting them if there was no pipe or all processes of the pipe succeeded?

The above code handles that - try it! Then read http://fishshell.com/docs/current/index.html#cartesian-product.

Reply:
Nice I did not know that. That is a good idea.
How would you suggest then to handle the `brace_color` which applies to those enclosing braces too. I tried to minimize the amount of code a user of a prompt has to do to get a decent `$pipestatus` addition to his/her prompt.

Reply:
>How would you suggest then to handle the brace_color which applies to those enclosing braces too

I'd also leave that to the calling prompt. This function should understand pipestatus, and print that. Anything that can be handled outside, like how to embed that pipestatus in the prompt, should be.

Reply:
If I understand correctly you are suggesting changing all the prompts' use of `__fish_print_pipestatus` from:
```
__fish_print_pipestatus "[" "] " "|" (set_color yellow) (set_color --bold yellow) $last_pipestatus
```
to
```
echo (set_color --bold yellow)"["(set_color normal)(__fish_print_pipestatus "|" (set_color green) (set_color --bold yellow) $last_pipestatus)(set_color --bold yellow)"]"(set_color normal)
```
And then to reduce the `__fish_print_pipestatus` function to not accept the left and right brace arguments.

If so then I am sorry but I do not agree because:
- In your case the braces will always appear regardless of whether the last command was a pipe. To remedy that would require the prompt to also do the same checks the function does.
- It complicates the goal of adding pipe status to a prompt and is more code duplication. 

In its current form the `__fish_print_pipestatus` function does not rule out to be used the way you suggest as you can pass empty string arguments for left and right brace.

Reply:
You know what? It's okay as-is.